runtime/vm/CRIUHelpers.cpp

```diff
@@ -675,6 +666,15 @@ runInternalJVMRestoreHooks(J9VMThread *currentThread, const char **nlsMsgFormat)
 	}
 #endif /* JAVA_SPEC_VERSION >= 20 */
 
+	if (J9_ARE_ALL_BITS_SET(vm->checkpointState.flags, J9VM_CRIU_IS_NON_PORTABLE_RESTORE_MODE)) {
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		vm->portLibrary->isCheckPointAllowed = FALSE;
+		vm->checkpointState.flags &= ~J9VM_CRIU_IS_CHECKPOINT_ALLOWED;
+		j9port_control(J9PORT_CTLDATA_CRIU_SUPPORT_FLAGS, OMRPORT_CRIU_SUPPORT_ENABLED | J9OMRPORT_CRIU_SUPPORT_FINAL_RESTORE);
+	}
+
+	TRIGGER_J9HOOK_VM_PREPARING_FOR_RESTORE(vm->hookInterface, currentThread);
```

Comment:
Could you provide some insights about moving this trigger event early while `jvmRestoreHooks()` hasn't started yet?

Reply:
There are a lot of issues with the existing error handling. I plan on doing another PR to address them all. In the meantime we need to make sure the JIT restore hooks always run because at checkpoint the JIT halts some threads, if the hook is not called, those threads remain blocked and the JVM hangs.

We werent seeing this problem before because werent correctly reporting criu_dump failures